### ISSUE TITLE
Terraform unable to detect ct provider

### ISSUE THREAD

**Issue by clee231** [coreos/terraform-provider-ct#38]
I am having an issue where the terraform provider isn't detected.

I have a providers.tf file defined like so:
```
provider "ct" {
  version = "0.3.1"
}

provider "local" {
  version = "~> 1.0"
  alias = "default"
}

provider "null" {
  version = "~> 1.0"
  alias = "default"
}

provider "template" {
  version = "~> 1.0"
  alias = "default"
}

provider "tls" {
  version = "~> 1.0"
  alias = "default"
}
```
After manually installing ct v0.3.1 and specifying the directory for all plugins of the project, I get the following:
```
$ terraform init -plugin-dir=.terraform/plugins/linux_amd64/

Initializing provider plugins...

Missing required providers.

The following provider constraints are not met by the currently-installed
provider plugins:

* ct (0.3.1)

Terraform can automatically download and install plugins to meet the given
constraints, but this step was skipped due to the use of -get-plugins=false
and/or -plugin-dir on the command line.

If automatic installation is not possible or desirable in your environment,
you may manually install plugins by downloading a suitable distribution package
and placing the plugin's executable file in one of the directories given in
by -plugin-dir on the command line, or in the following directory if custom
plugin directories are not set:
    terraform.d/plugins/linux_amd64
```
The structure of the plugin folder is as follows:
```
$ tree .terraform
.terraform/
├── plugin_path
└── plugins
    └── linux_amd64
        ├── terraform-provider-ct_v0.3.1_x4
        ├── terraform-provider-local_v1.2.0_x4
        ├── terraform-provider-null_v1.0.0_x4
        ├── terraform-provider-template_v1.0.0_x4
        └── terraform-provider-tls_v1.2.0_x4

2 directories, 6 files
```
I notice on the [terraform documention for 3rd party plugins](https://www.terraform.io/docs/configuration/providers.html#third-party-plugins) that the naming convention doesn't include the `_x4` part of the current filenames, however removing that part of the name still yields the same result.

I am currently using Terraform v0.11.13.  Am I missing something?

**Comment by dghubble** [comment]
I doubt Terraform will infer that by `.terraform/plugins/linux_amd64/` you mean within your home directory. Try a full path. Also, you probably don't need to specify, the default should be fine.

**Comment by dghubble** [comment]
You also don't neccessarily need the os/arch specific directories either. And drop the `_x4`, not sure why that's there.

Here's a related guide with examples I've [written](https://typhoon.psdn.io/topics/maintenance/#upgrade-terraform-provider-ct).

**Comment by clee231** [comment]
That path was actually just a relative path for a folder in the current directory.
I originally tried it without the `-plugin-dir=` flag, but it kept giving the the "Provider "ct" not available for installation." error, despite having the provider in my `~/.terraform.d/plugins` folder.
By explicitly using the `-plugin-dir`, I figured I could ensure all the providers were in the folder I specified explicitly.

---
Even with an absolute path, it does did not detect this provider for some reason. Renaming the provider with or without the `_x4` suffix did not fix the issue.

What's weird is that I tried this same setup on a different machine and it worked using the default methods (vanilla `terraform init` and placing the providers within the `$HOME/.terraform.d/plugins` directory).  It must be a weird configuration I have on this machine or something.

I have a slight suspicion that this is more of an issue with terraform, rather than a terraform-provider-ct issue, so I'll probably submit an issue with them to hopefully gain some more insight on the situation.
Thanks for your help!
